test/int/deployment_test.go

```diff
@@ -1810,3 +1863,28 @@ func mergedAdvancedDeployment(
 
 	return
 }
+
+func performUpdate[T any](ctx context.Context, timeout time.Duration, key client.ObjectKey, mutator func(*T)) *T {
+	obj, err := retryUpdateOnConflict(ctx, k8sClient, key, mutator)
+	Expect(err).To(BeNil())
+
+	clientObj := any(obj).(api.AtlasCustomResource)
+	Eventually(func(g Gomega) bool {
+		return resources.CheckCondition(k8sClient, clientObj, api.TrueCondition(api.ReadyType), validateDeploymentUpdatingFunc(g))
+	}).WithTimeout(timeout).WithPolling(interval).Should(BeTrue())
+
+	return obj
+}
+
+func retryUpdateOnConflict[T any](ctx context.Context, k8s client.Client, key client.ObjectKey, mutator func(*T)) (*T, error) {
```

Comment:
I have the feeling I have seen this function before. Also, it seems to be generic enough to be used in many cases. Shouldn't it be moved to the help utility package?

Reply:
I would do this on demand 👍

Reply:
https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1625 here we go :-)